code/modules/projectiles/projectile/magic.dm

```diff
@@ -406,11 +406,21 @@
 	. = ..()
 	if(isliving(A) && locker_suck)
 		var/mob/living/M = A
-		if(M.anti_magic_check())			// no this doesn't check if ..() returned to phase through do I care no it's magic ain't gotta explain shit
-			M.visible_message("<span class='warning'>[src] vanishes on contact with [A]!</span>")
-			return PROJECTILE_DELETE_WITHOUT_HITTING
+		if(M.anti_magic_check())
+			M.visible_message("<span class='warning'>[src] brutally hits [A]!</span>", "<span class='boldwarning'>You are hit by a a locker!")
+			var/hit_zone = pick(BODY_ZONE_HEAD,BODY_ZONE_CHEST,BODY_ZONE_PRECISE_GROIN) //hit in the balls by a locker
+			M.apply_damage(20, BRUTE, hit_zone, M.run_armor_check(hit_zone, MELEE))
+
+			//launch very hard
+			var/atom/throw_target = get_edge_target_turf(M, get_dir(src, get_step_away(M, src)))
+			M.safe_throw_at(throw_target, 300, 2, null, TRUE, FALSE, null, MOVE_FORCE_OVERPOWERING)
+
+			playsound(M, 'sound/weapons/smash.ogg', 50)
+			return PROJECTILE_PIERCE_HIT
```

Comment:
Testing this, it results in your projectile passing through walls and other dense objects for about a tile. This results in your locker phasing through walls and runtiming.

`[04:29:03] Runtime in atoms_movable.dm,363: Cannot execute null.Exited().
  proc name: Move (/atom/movable/Move)
  src: the locker bolt (/obj/projectile/magic/locker)
  src.loc: the floor (161,163,2) (/turf/open/floor/plasteel)
  call stack:
  the locker bolt (/obj/projectile/magic/locker): Move(the floor (161,163,2) (/turf/open/floor/plasteel), 1, 0)
  the locker bolt (/obj/projectile/magic/locker): Move(the floor (161,163,2) (/turf/open/floor/plasteel), 1, 0)
  the locker bolt (/obj/projectile/magic/locker): pixel move(1, 0)
  the locker bolt (/obj/projectile/magic/locker): process(0.1)
  Projectiles (/datum/controller/subsystem/processing/projectiles): fire(0)
  Projectiles (/datum/controller/subsystem/processing/projectiles): ignite(0)
  Master (/datum/controller/master): RunQueue()
  Master (/datum/controller/master): Loop()
  Master (/datum/controller/master): StartProcessing(0)
`

Reply:
I embraced the wall glitching, but runtimes are bad so I’ll look into that

Reply:
Don't embrace the wall glitching, it looks dreadful. It's probably runtiming as a result if anything but you should have recognized this probably doesn't make sense and shouldn't work like this.

Reply:
Sometimes you gotta stop asking questions and just accept unpredictable behaviour, as long as it’s amusing and doesn’t hurt player experience. 

I obviously didn’t notice the runtime though, the clippings been there since bullet penetration got reworked. I’ll fix that and probably clipping along with it

Reply:
I think I fixed the runtimes? I switched flags and I can't reproduce it, but to be fair I never got a runtime in the first place